### PR TITLE
Pull Request

### PR DESCRIPTION
**Pull request by Sunbread**
@micromaomao , Please review these code, fix wrong codes and merge them to master branch.

### CODE REVIEW

File: bf.js [hunk 1]
```diff
@@ -117,6 +117,11 @@ machine.prototype.run = function(tap, doHyper, onStop){
         onStop();
         return;
     }
+    if(this.program.match(new RegExp("\\[","g")).length != this.program.match(new RegExp("\\]","g")).length){
```
Comment (micromaomao) [inline bf.js]: Bored boring

Comment (micromaomao) [inline bf.js]: use `match(/xx/g)`

File: bf.js [hunk 2]
```diff
@@ -195,6 +200,7 @@ machine.prototype.nextStep = function(){
             if(this.tap.get() == 0){
                 var mt = foundMatch("[", "]", izl, true);
                 if(mt === undefined){
+					this.stderr("\[ and \] can't match.", false);
```
Comment (micromaomao) [inline bf.js]: WHAT THE HELL is this indent...

File: bf.js [hunk 3]
```diff
@@ -204,6 +210,7 @@ machine.prototype.nextStep = function(){
             if(this.tap.get() != 0){
                 var mt = foundMatch("]", "[", izl, false);
                 if(mt === undefined){
+					this.stderr("\[ and \] can't match.", false);
```
Comment (micromaomao) [inline bf.js]: WHAT THE F**K YOU EDITOR ARE DOING..

File: bf.js [hunk 4]
```diff
@@ -218,6 +225,9 @@ machine.prototype.nextStep = function(){
 machine.prototype.stdout = function(out){
     this.oA.value += out;
 };
+machine.prototype.stderr = function(err, fatal){
+    this.oA.value += (fatal?"\[FATAL Error\] ":"\(Error: ") + out + (fatal?"":"\)");
```
Comment (micromaomao) [inline bf.js]: please use `stdout()`.

**Comment by micromaomao** [comment]
OK, Wait me, editing.

**Comment by Sunbread** [comment]
Those Tab are my problem...
The Notepad++ added so many Tab.

**Comment by Sunbread** [comment]
You just change every <Tab> to four spaces.

**Pull request merged**